### ISSUE TITLE
Safari: Download OTF Fails to create export file

### ISSUE THREAD

**Issue by FergusClare** [glyphr-studio/Glyphr-Studio-1#248]
Looks like the Sept 16 work may have inadvertently broken the .otf download when clicking export.  I spent a few hours making a font only have the download fail on me after attempting get the otf.

Machine: mac
OS. OSX
Date: 01/24/2017
Time: 10:30PM CST
Browser: Safari
Extensions: Ad Blocker (disabled on site)

Steps to Reproduce:
1. Open glyphr
2. Start new project
3. attempt to export .otf
4. Script runs to export (seen in blue box at top of screen)
5. Browser load bar below omnibox then stalls
6. Project file is missing from downloads and machine when using spotlight search

Sample webarchives and svg attached that I was able to obtain before refresh can be obtained via email.

**Comment by mattlag** [comment]
This is a Safari issue - Error message is:

```
TypeError: '[object BlobConstructor]' is not a constructor (evaluating 'new Blob([b],{type:c,endings:"native"})')
```

Confirmed working on Windows in Chrome / FireFox / Edge / Internet Explorer

**Comment by mattlag** [comment]
Hi @FergusClare 

Maybe you could help me out investigating this.  I only have access to Windows machines.  There is a version of Safari for Windows, but it's quite old (version 5.1.7 from 2012).

If you could do this, it may help me understand the problem:

* Go to http://opentype.js.org/
* Show web inspector (maybe Ctrl+Alt+I)
* in the console type: `font.download();`

Glyphr Studio relies on OpenType.js to write font files.  I'm trying to figure out if the error is on the OpenType.js side or on the Glyphr Studio side.

Thanks!

**Comment by FergusClare** [comment]
@mattlag 

Thanks so much for the quick reply and for getting me a copy of font.  I did as you suggested and received (!) Error: Can't create coverage table format 2 with the following results:

`fail — opentype.js:564
argument — opentype.js:571
Coverage — opentype.js:3742
makeLookup1 — opentype.js:6106
tableList — opentype.js:3708
(anonymous function) — opentype.js:3816
tableList — opentype.js:3708
LookupList — opentype.js:3810
makeGsubTable — opentype.js:6142
fontToSfntTable — opentype.js:8073
toArrayBuffer — opentype.js:1314
download — opentype.js:1332
Global Code
evaluateWithScopeExtension
_evaluateOn
_evaluateAndWrap
evaluate`

Attaching output of the .js file so you can look line by line if needed.  Happy to provide more testing/results and thanks again!

[openTypeJSError.txt](https://github.com/glyphr-studio/Glyphr-Studio-1/files/730598/openTypeJSError.txt)

**Comment by mattlag** [comment]
Yep, unfortunately this looks like OpenType.js is failing in Safari... i'll file a bug with them, maybe they can figure out what's wrong.

**Comment by mattlag** [comment]
On second look, this also fails in Chrome :-) If you could copy / paste this code (it's just their basic example from their GitHub page) into the Web Inspector at command line at http://opentype.js.org/ 

```
var notdefGlyph = new opentype.Glyph({
    name: '.notdef',
    unicode: 0,
    advanceWidth: 650,
    path: new opentype.Path()
});

var aPath = new opentype.Path();
aPath.moveTo(100, 0);
aPath.lineTo(100, 700);

var aGlyph = new opentype.Glyph({
    name: 'A',
    unicode: 65,
    advanceWidth: 650,
    path: aPath
});

var glyphs = [notdefGlyph, aGlyph];
var font = new opentype.Font({
    familyName: 'OpenTypeSans',
    styleName: 'Medium',
    unitsPerEm: 1000,
    ascender: 800,
    descender: -200,
    glyphs: glyphs});
font.download();
```

This does successfully trigger a download in Chrome.

**Comment by jeshyr** [comment]
Since it looks like @FergusClare isn't around and I use the same setup, here's the relevant bit. Everything works fine right up to that final line -

![progresscursor](https://user-images.githubusercontent.com/144218/27317256-72b11d1c-55c8-11e7-95db-ddc210a395e2.jpg)

**Comment by mattlag** [comment]
Thank you, i'll look into this.

**Comment by mattlag** [comment]
I believe this is fixed in v1.11 - let me know if not.

Also, I submitted a pull request to have this (potentially) fixed in opentype.js as well:
https://github.com/nodebox/opentype.js/pull/354

**Comment by Jolg42** [comment]
Merged in opentype.js master branch, and soon on npm.
